tests/unit/s2n_record_size_test.c

```diff
@@ -84,9 +84,139 @@ int main(int argc, char **argv)
     EXPECT_EQUAL(bytes_written, large_aligned_payload);
 
     /* Clean up */
+    conn->secure.cipher_suite->record_alg = &s2n_record_alg_null; /* restore mutated null cipher suite */
     EXPECT_SUCCESS(conn->secure.cipher_suite->record_alg->cipher->destroy_key(&conn->secure.server_key));
     EXPECT_SUCCESS(conn->secure.cipher_suite->record_alg->cipher->destroy_key(&conn->secure.client_key));
     EXPECT_SUCCESS(s2n_connection_free(conn));
 
+    /* Test s2n_record_max_write_payload_size() have proper checks in place */
+    {
+        struct s2n_connection *server_conn;
+        EXPECT_NOT_NULL(server_conn = s2n_connection_new(S2N_SERVER));
+
+        /* we deal with the default null cipher suite for now, as it makes reasoning
+         * about easier s2n_record_max_write_payload_size(), as it incur 0 overheads */
+        int size;
+        server_conn->max_outgoing_fragment_length = 1024;
+        EXPECT_SUCCESS(size = s2n_record_max_write_payload_size(server_conn));
+        EXPECT_EQUAL(size, 1024);
+
+        /* Trigger an overlarge payload by setting a maximum uint16_t value to max fragment length */
+        server_conn->max_outgoing_fragment_length = 65535;
+        /* Check that we have an error guard */
+        EXPECT_FAILURE_WITH_ERRNO(s2n_record_max_write_payload_size(server_conn), S2N_ERR_FRAGMENT_LENGTH_TOO_LARGE);
+
+        /* trigger a payload that is under the limits */
+        server_conn->max_outgoing_fragment_length = 0;
+        EXPECT_FAILURE_WITH_ERRNO(s2n_record_max_write_payload_size(server_conn), S2N_ERR_FRAGMENT_LENGTH_TOO_SMALL);
+
+        /* Test boundary cases */
+
+        /* This is the theorical maximum mfl allowed */
+        server_conn->max_outgoing_fragment_length = 16384;
```

Comment:
this is defined as `S2N_TLS_MAXIMUM_RECORD_LENGTH`

can we use constants where appropriate, please?

Reply:
I'm spelled numbers explicitly because I'm using them to check against s2n numbers, in case they become a little echo chamber. Let me see where it make sense to clean them up with constants.

Reply:
make sense, thanks